mslib/mscolab/server.py

```diff
@@ -89,6 +102,37 @@ def verify_pw(username, password):
         return authfunc(username, password)
 
 
+def send_email(to, subject, template):
+    msg = Message(
+        subject,
+        recipients=[to],
+        html=template,
+        sender=APP.config['MAIL_DEFAULT_SENDER']
+    )
+    try:
+        mail.send(msg)
+    except IOError:
+        logging.debug(f"Can't send email to {to}")
```

Comment:
Do not use f-strings. Use % for performance reasons etc.

Reply:
My personal timeit performance check turned out, at best, inconclusive and at worst show c-strings being worse than f-strings at high numbers, but the difference is negligible.
I would recommend using what is most readable, which is f-strings.

Reply:
we have already other f-strings for logging.

Reply:
f-formatting is almost always faster than c-style formatting due to some particularly interesting implementation shenanigans. That is not the point here, however.

The idea here is that the formatting will *not take place at all* in case the logging level is above debug. Using an f-string always does the formatting even if nothing will be output. That can have an detrimental effect on performance *with logging disabled*.
Thus it is good style to *always* employ conditional formatting, which by necessity of backwards-compatability, is c-style formatting. For info/warn/error, which are basically always dumped or where the formatted string is needed anyway (e.g. for a popup message), one may deviate at discretion. 

There is a "style" parameter, which can be triggered on a *by call* basis to use other formatters (e.g. the {} one). See
https://docs.python.org/3/howto/logging-cookbook.html#formatting-styles
This seems very cumbersome for little gain.
f-strings cannot be used as the decision whether to format cannot be controlled by the logging class.

Reply:
> we have already other f-strings for logging.

Then these need to be replaced as well.

Reply:
this is than another refactoring, fixed that one

Reply:
Fine for me if it is fixed here.
We had that discussion - and fixes - in the past.
E.g.
https://github.com/Open-MSS/MSS/issues/536

Reply:
I remember ;)